src/Dotnet.Installer.Core/Services/Implementations/ManifestService.Private.cs

```diff
@@ -26,30 +26,35 @@ private static async Task<List<Component>> LoadLocal(CancellationToken cancellat
     private static async Task<List<Component>> LoadRemote(bool includeUnsupported = false,
         CancellationToken cancellationToken = default)
     {
-        var supportedJsonFilePath = Path.Join("/", "snap", "dotnet-manifest", "current", "supported.json");
-        if (!File.Exists(supportedJsonFilePath))
+        var filesToRead = new List<string>
         {
-            throw new ApplicationException("dotnet-manifest snap is not installed. Please run 'sudo snap install dotnet-manifest'.");
-        }
+            Path.Join("/", "snap", "dotnet-manifest", "current", "supported.json")
+        };
 
-        var contentStream = File.OpenRead(supportedJsonFilePath);
-        var components = await JsonSerializer.DeserializeAsync<List<Component>>(
-            contentStream, JsonSerializerOptions, cancellationToken);
+        if (includeUnsupported)
+            filesToRead.Add(Path.Join("/", "snap", "dotnet-manifest", "current", "unsupported.json"));
 
-        if (components is null)
+        var components = new List<Component>();
+        foreach (var contentStream in filesToRead.Select(File.OpenRead))
         {
-            throw new ApplicationException("Could not read the supported.json file.");
-        }
+            var currentComponents = await JsonSerializer.DeserializeAsync<List<Component>>(
+                contentStream,
+                JsonSerializerOptions,
+                cancellationToken: cancellationToken);
 
-        // TODO Add unsupported flag support
+            if (currentComponents is not null)
```

Comment:
**question:** How do get deserialization errors handled now? It looks to me like they are just swallowed or cause the application to crash.

**note:** I am aware that these errors should be rare, because we have a lot of control over the environment.